src/orion/benchmark/task/quadratics.py

```diff
@@ -0,0 +1,505 @@
+    # # > (1.0, 0.0)
+    # r_score, p_value = spearmanr(y_low, y_high)
+    # logger.debug(f"Spearman: {r_score=}, {p_value=}")
+
+    d = distance(task_a, task_b)
+    # Here for this particular task we know the most different values possible.
+    max_dist = max(
+        distance(task_a, QuadraticsTask.task_low()),
+        distance(task_a, QuadraticsTask.task_high()),
+        distance(task_b, QuadraticsTask.task_low()),
+        distance(task_b, QuadraticsTask.task_high()),
+    )
+    assert 0 <= d <= max_dist
+    logger.debug(f"{task_a=} {task_b=}, {d=}, {max_dist=}")
+    return 1 - d / max_dist
+    # return 1 / d if d != 0 else 1.0
+    # return 1 / d
+
+    # spearman_r, spearman_p = spearmanr(y_a, y_b)
+    # pearson_r, pearson_p = pearsonr(y_a, y_b)
+    # assert False, (spearman_r, spearman_p, pearson_r, pearson_p)
+    # return r
+
+    # max_distance = 0
+    # y_max = QuadraticsTask.task_high()(x_a)
+    # y_min = QuadraticsTask.task_low()(x_a)
+    # max_distance = np.abs(y_max - y_min).sum() / n
+    # distance = np.abs(y_a - y_b).sum() / n
+
+    # similarity = distance / max_distance
+    # return similarity
+
+
+@distance.register
+def distance_between_quadratics_task(
+    task_a: QuadraticsTask, task_b: QuadraticsTask, n: int = 100, **unused_kwargs
+) -> float:
+    """ TODO: 'distance' between two quadratic functions.
+
+    For now using something simple
+    """
+    distance = (
+        abs(task_a.a2 ** 2 - task_b.a2 ** 2)
+        + abs(task_a.a1 - task_b.a1)
+        + abs(task_a.a0 - task_b.a0)
+    )
+    return distance
+
+
+# @similarity.register
+# def similarity(
+#     task_a: QuadraticsTask, task_b: QuadraticsTask, n: int = 100, **unused_kwargs
+# ) -> float:
+#     xs = task_a.sample_np(n=n)
+#     y_a: np.ndarray = task_a.call(xs)
+#     y_b: np.ndarray = task_b.call(xs)
+#     # logger.debug(y_a, y_b)
+#     # y_a -= y_a.mean()
+#     # y_a /= y_a.std()
+
+#     # y_b -= y_b.mean()
+#     # y_b /= y_b.std()
+
+#     if (y_a == y_b).all():
+#         return 1.0
+
+#     from scipy.stats import pearsonr, wilcoxon
+#     from warmstart.utils.utils import sigmoid
+
+#     distance = np.abs(y_a - y_b).sum() / n
+#     similarity = 1 / (1 + distance)
+#     ret
+#     # x_mean = xs.mean(0)
+#     # distance_2 = (task_a.a2 - task_b.a2) * (x_mean ** 2) + abs(task_a.a1 - task_b.a1) * x_mean + abs(task_a.a0 - task_b.a0)
+#     # assert False, (distance, distance_2)
+#     wilcoxon_score, wilcoxon_p_value = wilcoxon(y_a, y_b)  # zero_method="zsplit")
+#     pearson_r_score, pearson_p_value = pearsonr(y_a, y_b)
+#     assert False, (similarity, wilcoxon_p_value, wilcoxon_score / n, pearson_r_score, pearson_p_value)
+#     return r_score
+
+
+@dataclass
+class QuadraticsTaskWithContextHparams(QuadraticsTaskHParams):
+    # This adds these entries to the space, but they aren't used in the
+    # quadratic equation above. (`__call__` of QuadraticsTask.)
+    a2: float = uniform(0.1, 10.0)
```

Comment:
I don't understand what is the difference. Aren't these already set to uniform(0.1, 10.0) in the parent class?

Reply:
Oh, that's the Hparams class. So here instead of x being the HPs it's a?

Reply:
In a sense yeah, The points contain both the values for `x` and the coefficients of the quadratic (the "context" task in ABLR)